W-22013386-namespace-in-dw-script-CP (#612)
mulesoft/docs-code-builder: modules/ROOT/pages/int-configure-dw-expressions.adoc

* <<map-with-ai, Map with AI>>
* <<map-and-transform-with-ai, Map and Transform with AI>>

[[namespaces-in-generated-dataweave]]

> === Namespaces in Generated DataWeave

Reviewer: ```suggestion
=== Namespaces in Generated DataWeave Scripts
```
I added this because DW is the language, but what gets generates is a DW script (if I understood correctly).

Author: OK! Thanks!

> If your input or output metadata includes namespace URIs, Anypoint Code Builder updates the generated DataWeave script for visual mapping to XML and similar targets. It adds `ns` directives to the script header for each URI that needs a declaration and maps fields using qualified names with a generated prefix. Prefixes are unique in the script (such as `ns0`, `ns1`), and an existing `ns` line is reused if it binds the same URI.

Reviewer: ```suggestion
If your input or output metadata includes namespace URIs, Anypoint Code Builder updates the generated DataWeave script for visual mapping to XML and similar targets. The IDE adds `ns` directives to the script header for each URI that needs a declaration and maps fields using qualified names with a generated prefix. Prefixes are unique in the script (such as `ns0`, `ns1`), and Anypoint Code Builder reuses any existing `ns` line if it binds the same URI.
```

Author: OK! Thanks!

> Namespace handling applies to AI-assisted mapping and manual drag mapping in *Transformation Builder*. It also applies to other flows that generate script from metadata, for MIME types where the type system exposes namespaces. Supply accurate xref:int-manage-custom-metadata.adoc[custom metadata] or sample data so element namespaces are present in the structure the builder uses.

Reviewer: ```suggestion
Namespace handling applies to AI-assisted mapping and manual drag mapping in *Transformation Builder*. It also applies to other flows that generate scripts from metadata, for MIME types where the type system exposes namespaces. Provide accurate xref:int-manage-custom-metadata.adoc[custom metadata] or sample data so element namespaces are present in the structure the builder uses.
```
If Transformation Builder is not a UI element, consider removing the bold formatting.

Author: Yes! Transform Builder is the name of the UI ( a sort of modal) in which the user can transform date.
